src/sentry/static/sentry/app/components/search/sources/helpSource.jsx

```diff
@@ -0,0 +1,173 @@
+import {debounce} from 'lodash';
+import {withRouter} from 'react-router';
+import PropTypes from 'prop-types';
+import React from 'react';
+import algoliasearch from 'algoliasearch';
+import styled from 'react-emotion';
+
+import {
+  ALGOLIA_APP_ID,
+  ALGOLIA_READ_ONLY,
+  ALGOLIA_DOCS_INDEX,
+  ALGOLIA_ZENDESK_INDEX,
+} from 'app/constants';
+import parseHtmlMarks from 'app/utils/parseHtmlMarks';
+import withLatestContext from 'app/utils/withLatestContext';
+
+/**
+ * Use unique markers for highlighting so we can parse these into fuse-style
+ * indicidies.
+ */
+const HIGHLIGHT_TAGS = {
+  highlightPreTag: '<algolia-highlight-0000000000>',
+  highlightPostTag: '</algolia-highlight-0000000000>',
+};
+
+const SNIPPET_LENGTH = 260;
+
+class HelpSource extends React.Component {
+  static propTypes = {
+    // search term
+    query: PropTypes.string,
+
+    /**
+     * Render function that passes:
+     * `isLoading` - loading state
+     * `allResults` - All results returned from all queries: [searchIndex, model, type]
+     * `results` - Results array filtered by `this.props.query`: [searchIndex, model, type]
+     */
+    children: PropTypes.func.isRequired,
+  };
+
+  constructor(props, ...args) {
+    super(props, ...args);
+    this.state = {
+      loading: false,
+      results: null,
+    };
+
+    this.algolia = algoliasearch(ALGOLIA_APP_ID, ALGOLIA_READ_ONLY);
+  }
+
+  componentDidMount() {
+    if (typeof this.props.query !== 'undefined') {
```

Comment:
why do we use a typeof here?

Reply:
Not sure, this was actually copied over from your `DocSearch` :0

Reply:
the `undefined` global can be changed to something else.

Reply:
Yeah I've always done this out of habit because technically it was possible to change the value of `undefined`, but I'm not even sure if that's still possible